### DISCUSSION TITLE
How can I pass preview API urls to my client's env variables automatically?

### DISCUSSION THREAD

**Discussion by Jakemangan** [vercel/community#68] [Help] [answered]
Forgive me if this is a simple question but I can't seem to find any information on it.

When I deploy my API to a preview in Vercel, the latest commit hash gets appended to the URL of the API. Which means that each time I deploy my API it's going to have a different URL.

Because of this, if I set my client to use an API URL, that URL is then going to be out of date when I next deploy my API. I'd have to then manually go into my client's deployment and reconfigure the environment variable for my API URL to be the latest.

That sounds quite cumbersome, is there anyway to automate this? Or am I stuck with using only the project's production URL in which case I need to merge everything into master/main and then deploy that way (which means I couldn't have my client access a feature branch API unless I manually configure).

**Comment by leerob** [comment, marked as answer]
Please see the docs here about how URLs are created for Preview environments: https://vercel.com/docs/concepts/deployments/automatic-urls
